server/service/IndividualServicesService.js

```diff
@@ -11084,6 +11086,54 @@ exports.regardControllerAttributeValueChange = function (url, body, user, origin
   });
 }
 
+// Map of per-mountname locks
+const locks = new Map();
+const queueCounts = new Map(); // track queue depth per mountname
+
+/**
+ * Ensures only one update per mountname runs at a time.
+ * id String mountname
+ * alarmBody V1_regarddevicealarm_body
+ * fn Function async function that performs the update
+ * returns {Promise<any>} result of fn
+ */
+async function withLock(id, alarmBody, fn) {
+  if (!locks.has(id)) {
+    locks.set(id, Promise.resolve());
+    queueCounts.set(id, 0);
+  }
+  const prev = locks.get(id);
+  queueCounts.set(id, (queueCounts.get(id) || 0) + 1);
+  const queueSize = queueCounts.get(id);
+
+  logAlarmNotificationUpdate(`***************************************START - Processing alarm for Mountname=${id}***************************************`);
+  logAlarmNotificationUpdate(`Mountname=${id} queued update. Queue size now: ${queueSize}`);
+  logAlarmNotificationUpdate(`Alarm Mountname=${id} to be updated: ${alarmBody}`);
+  logger.info(`Mountname=${id} queued update. Queue size now: ${queueSize}`);
+
+  let release;
+  const p = new Promise((res) => (release = res));
+  locks.set(id, prev.then(() => p));
+
+  try {
+    const start = Date.now();
+    const result = await fn();
```

Comment:
Check if fn() hangs or throws an error, it could block the queue for that id.
If such scenarios happen , add a timeout wrapper or error isolation, something like this, 
```js
const timeout = (ms) => new Promise((_, reject) => setTimeout(() => reject(new Error("Timeout")), ms));
const result = await Promise.race([fn(), timeout(10000)]); // 10s timeout
```